client/components/search/index.jsx

```diff
@@ -293,42 +309,52 @@ const Search = React.createClass( {
 					}
 					aria-controls={ 'search-component-' + this.state.instanceId }
 					aria-label={ i18n.translate( 'Open Search', { context: 'button label' } ) }>
-				<Gridicon icon="search" className={ 'search-open__icon' + ( this.props.dir ? ' ' + this.props.dir : '' ) }/>
+				<Gridicon icon="search" className={ gridIconClass }/>
 				</div>
-				<input
-					type="search"
-					id={ 'search-component-' + this.state.instanceId }
-					className={ 'search__input' + ( this.props.dir ? ' ' + this.props.dir : '' ) }
-					placeholder={ placeholder }
-					role="search"
-					value={ searchValue }
-					ref="searchInput"
-					onChange={ this.onChange }
-					onKeyUp={ this.keyUp }
-					onKeyDown={ this.keyDown }
-					onFocus={ this.onFocus }
-					onBlur={ this.onBlur }
-					disabled={ this.props.disabled }
-					aria-hidden={ ! isOpenUnpinnedOrQueried }
-					autoCapitalize="none"
-					dir={ this.props.dir }
-					{ ...autocorrect } />
-				{ ( searchValue || this.state.isOpen ) ? this.closeButton() : null }
+				<div className={ fadeDivClass }>
+					<input
+						type="search"
+						id={ 'search-component-' + this.state.instanceId }
+						className={ inputClass }
+						placeholder={ placeholder }
+						role="search"
+						value={ searchValue }
+						ref="searchInput"
+						onChange={ this.onChange }
+						onKeyUp={ this.keyUp }
+						onKeyDown={ this.keyDown }
+						onFocus={ this.onFocus }
+						onBlur={ this.onBlur }
+						disabled={ this.props.disabled }
+						aria-hidden={ ! isOpenUnpinnedOrQueried }
+						autoCapitalize="none"
+						dir={ this.props.dir }
+						{ ...autocorrect }
+					/>
+				</div>
+				{ this.closeButton() }
 			</div>
 		);
 	},
 
 	closeButton: function() {
-		return (
-			<span
-				onTouchTap={ this.closeSearch }
-				tabIndex="0"
-				onKeyDown={ this.closeListener }
-				aria-controls={ 'search-component-' + this.state.instanceId }
-				aria-label={ i18n.translate( 'Close Search', { context: 'button label' } ) }>
-			<Gridicon icon="cross" className={ 'search-close__icon' + ( this.props.dir ? ' ' + this.props.dir : '' ) } />
-			</span>
-		);
+		const gridIconClass = classNames( 'search-close__icon', this.props.dir );
```

Comment:
Given the comment above about adding this.props.dir just once, I'd get rid of the gridicon class in favor of `search__close-icon`. Also note the class naming guidelines, `search-close` is incorrect.